File: bddtests/src/main/kotlin/mocking/tpp/models/Registration.kt
package mocking.tpp.models

import javax.xml.bind.annotation.XmlAccessType
import javax.xml.bind.annotation.XmlAccessorType
import javax.xml.bind.annotation.XmlElement

@XmlAccessorType(XmlAccessType.FIELD)
data class Registration(
        @field:XmlElement(name = "PatientAccess") var patientAccess: MutableCollection<PatientAccess> = mutableListOf()
)

File: xamarinintegrationtests/NHSOnline.HttpMocks/Tpp/Models/Registration.cs
using System;
using System.Collections.ObjectModel;
using System.Xml.Serialization;

namespace NHSOnline.HttpMocks.Tpp.Models
{
    [Serializable]
    public sealed class Registration
    {
        [XmlElement("PatientAccess")]
        public Collection<PatientAccess> PatientAccess { get; } = new Collection<PatientAccess>();
    }
}